R/mapping_using_bigQuery_drugExposure.R

```diff
@@ -0,0 +1,116 @@
+# Load necessary libraries
+library(dplyr)
+library(DBI)
+library(readr)
+library(AMR)  # For drug name mapping
+
+# Load data from CSV files
+drug_exposure <- read_csv("../data/OMOP/cleaned_drug_exposure.csv")
+concept <- read_csv("../data/OMOP/cleaned_concept.csv")
+
+
+
+# Print the column names to check if they are correct
+print(colnames(concept))
+# Print the cleaned data to verify everything looks correct
+print(head(concept))
+
+
+# Ensure the relevant columns are available in drug_exposure
+print(colnames(drug_exposure))
+print(head(drug_exposure))  
+
+
+#drug_exposure <- drug_exposure %>%
+#  left_join(concept, by = c("drug_concept_id" = "concept_id")) %>%
+#  rename(drug_name = concept_name) %>%
+#  select(-domain_id, -vocabulary_id, -concept_class_id, -standard_concept, -concept_code, 
+#         -valid_start_date, -valid_end_date, -invalid_reason)
+
+#Mapping using left join
+drug_exposure <- drug_exposure %>%
+  left_join(concept, by = c("drug_concept_id" = "concept_id")) %>%
+  rename(drug_name = concept_name) %>%  # Rename concept_name to drug_name
+  mutate(route = NA)  # Since route_concept_id is NA.
+
+# Remove unnecessary columns from concept table
+drug_exposure <- drug_exposure %>%
+  select(-domain_id, -vocabulary_id, -concept_class_id, -standard_concept, 
+         -concept_code, -valid_start_date, -valid_end_date, -invalid_reason)
+
+
+# Map 'dose_unit_source_value' directly as units (since 'dose_unit_concept_id' is missing)
+if ("dose_unit_source_value" %in% colnames(drug_exposure)) {
+  drug_exposure <- drug_exposure %>%
+    mutate(units = dose_unit_source_value)  # Map 'dose_unit_source_value' to units
+} else {
+  drug_exposure$units <- NA  # If 'dose_unit_source_value' is missing, set units as NA
+}
+
+# Map drug_exposure fields to RAMSES fields
+omop_to_ramses <- drug_exposure %>%

```

Comment:
```suggestion
omop_to_ramses_drug_prescriptions <- drug_exposure %>%
```
As we will be producing other scripts for mapping to the other ramses tables, it would be good to rename this data frame to match the ramses table that is being mapped. This name would then need to be change in the rest of the script.